web-admin/src/features/public-urls/form-utils.ts

```diff
@@ -43,3 +44,40 @@ export function convertDateToMinutes(date: string) {
   const diff = future.getTime() - now.getTime();
   return Math.floor(diff / 60000);
 }
+
+/**
+ * Returns the serialized *sanitized* `state` for the current dashboard.
+ * It removes any filters and any pivot chips that refer to those filters.
+ * This ensures we do not leak hidden filters to the URL recipient.
+ */
+export function getSanitizedDashboardStateParam(
+  dashboard: MetricsExplorerEntity,
+  metricsViewFields: string[] | undefined,
+): string {
+  // If no metrics view fields are specified, everything is visible, and there's no need to sanitize
+  if (!metricsViewFields) return getProtoFromDashboardState(dashboard);
+
+  const sanitizedDashboardState = {
+    ...dashboard,
```

Comment:
Would be good to add the exact things we want rather than filter out things we do not want. This could lead to leaked data if a new feature doesnt add this filter.

For example these are not filtered right now (not exhaustive, would need a deeper dive to get full list).
1. List of selected measures and dimensions.
2. Leaderboard measure.
3. TTD measure name if present.

Reply:
True, adding state explicitly makes it easier to get right. Just updated.